src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/CsvDataExtractionService.kt

```diff
@@ -51,11 +51,8 @@ class CsvDataExtractionService(
   }
 
   private fun writeCourtBookingsToCsv(events: Stream<VideoBookingEvent>, csvOutputStream: OutputStream): Int {
-    val locationsByPrisonCode = mutableMapOf<String, List<Location>>()
-
     val courtEvents = events
-      .peek { locationsByPrisonCode.getOrPut(it.prisonCode) { locationsInsidePrisonClient.getNonResidentialAppointmentLocationsAtPrison(it.prisonCode) } }
-      .map { CourtBookingEvent(it, locationsByPrisonCode) }
+      .map { CourtBookingEvent(it, locationsInsidePrisonClient.getNonResidentialAppointmentLocationsAtPrison(it.prisonCode).toSet()) }
```

Comment:
The locations endpoint is cached so it is safe to call here.  The previous code was done before we had any caching.

Also peek should not be used here as it is not a terminating operation where as map is on a Sequence.